zeppelin-server/src/main/java/org/apache/zeppelin/service/NotebookService.java

```diff
@@ -279,6 +279,22 @@ public Note importNote(String notePath,
     }
   }
 
+  /**
+   * Return true only when paragraph execution finished, it could end with succeed or error due to user code.
+   * Return false when paragraph execution fails due to zeppelin internal issue.
+   * @param noteId
+   * @param paragraphId
+   * @param title
+   * @param text
+   * @param params
+   * @param config
+   * @param failIfDisabled
+   * @param blocking
+   * @param context
+   * @param callback
+   * @return
```

Comment:
should we move "return true only when ..." here and at beginning something like "executes given paragraph"...?

Reply:
javadoc is updated